File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/Source/DKImagePickerController/View/Cell/DKAssetGroupDetailImageCell.m
//
//  DKAssetGroupDetailImageCell.m
//  DKImagePickerControllerDemo_OC
//
//  Created by 赵铭 on 2017/6/26.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "DKAssetGroupDetailImageCell.h"
#import "DKImageResource.h"
@interface DKImageCheckView()
@property (nonatomic, strong) UIImageView * checkImageView;
@property (nonatomic, strong) UILabel * checkLabel;
@end

@implementation DKImageCheckView
- (UIImageView *)checkImageView{
    if (!_checkImageView) {
        _checkImageView = [[UIImageView alloc] initWithImage:[[DKImageResource checkedImage] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]];
    }
    return _checkImageView;
}

- (UILabel *)checkLabel{
    if (!_checkLabel) {
        _checkLabel = [[UILabel alloc] init];
        _checkLabel.textAlignment = NSTextAlignmentRight;
    }
    return _checkLabel;
}

- (instancetype)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame:frame]) {
        
        [self addSubview:self.checkImageView];
        [self addSubview:self.checkLabel];
    }
    return self;
}

- (void)layoutSubviews{
    [super layoutSubviews];
    self.checkImageView.frame = self.bounds;
    self.checkLabel.frame = CGRectMake(0, 5, self.bounds.size.width - 5, 20);
}

@end



@interface DKAssetGroupDetailImageCell()
@property (nonatomic, strong) UIImageView * thumbnailImageView;
@property (nonatomic, strong) DKImageCheckView * checkView;
@end

@implementation DKAssetGroupDetailImageCell
+ (NSString *)cellReuseIdentifier{
    return @"DKImageAssetIdentifier";
}
- (instancetype)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame:frame]) {
        self.thumbnailImageView.frame = self.bounds;
        self.thumbnailImageView.backgroundColor = [UIColor yellowColor];
        self.thumbnailImageView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        [self.contentView addSubview:self.thumbnailImageView];
        
        self.checkView = [DKImageCheckView new];
        self.checkView.frame = self.bounds;
        self.checkView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        self.checkView.checkImageView.tintColor = nil;
        self.checkView.checkLabel.font = [UIFont boldSystemFontOfSize:14];
        self.checkView.checkLabel.textColor = [UIColor whiteColor];
        [self.contentView addSubview:self.checkView];

    }
    return self;
}

- (UIImageView *)thumbnailImageView{
    if (!_thumbnailImageView) {
        _thumbnailImageView = [[UIImageView alloc] init];
        _thumbnailImageView.contentMode = UIViewContentModeScaleAspectFill;
        _thumbnailImageView.clipsToBounds = YES;
    }
    return _thumbnailImageView;
}
- (void)setThumbnailImage:(UIImage *)thumbnailImage{
    if (self.thumbnailImage != thumbnailImage) {
        super.thumbnailImage = thumbnailImage;
        self.thumbnailImageView.image = thumbnailImage;
    }
}

- (void)setIndex:(NSInteger)index{
    super.index = index;
    self.checkView.checkLabel.text = [NSString stringWithFormat:@"%ld", self.index + 1];
}

- (void)setSelected:(BOOL)selected{
    super.selected = selected;
    self.checkView.hidden = !selected;
}
/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: DKImagePickerController/View/Cell/DKAssetGroupDetailImageCell.swift
//
//  DKAssetGroupDetailImageCell.swift
//  DKImagePickerController
//
//  Created by ZhangAo on 07/12/2016.
//  Copyright © 2016 ZhangAo. All rights reserved.
//

import UIKit

class DKAssetGroupDetailImageCell: DKAssetGroupDetailBaseCell {
    
    class override func cellReuseIdentifier() -> String {
        return "DKImageAssetIdentifier"
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.thumbnailImageView.frame = self.bounds
        self.thumbnailImageView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        self.contentView.addSubview(self.thumbnailImageView)
        
        self.checkView.frame = self.bounds
        self.checkView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        self.checkView.checkImageView.tintColor = nil
        self.checkView.checkLabel.font = UIFont.boldSystemFont(ofSize: 14)
        self.checkView.checkLabel.textColor = UIColor.white
        self.contentView.addSubview(self.checkView)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    class DKImageCheckView: UIView {
        
        internal lazy var checkImageView: UIImageView = {
            let imageView = UIImageView(image: DKImageResource.checkedImage().withRenderingMode(.alwaysTemplate))
            return imageView
        }()
        
        internal lazy var checkLabel: UILabel = {
            let label = UILabel()
            label.textAlignment = .right
            
            return label
        }()
        
        override init(frame: CGRect) {
            super.init(frame: frame)
            
            self.addSubview(checkImageView)
            self.addSubview(checkLabel)
        }
        
        required init?(coder aDecoder: NSCoder) {
            fatalError("init(coder:) has not been implemented")
        }
        
        override func layoutSubviews() {
            super.layoutSubviews()
            
            self.checkImageView.frame = self.bounds
            self.checkLabel.frame = CGRect(x: 0, y: 5, width: self.bounds.width - 5, height: 20)
        }
        
    } /* DKImageCheckView */
    
    override var thumbnailImage: UIImage? {
        didSet {
            self.thumbnailImageView.image = self.thumbnailImage
        }
    }
    override var index: Int {
        didSet {
            self.checkView.checkLabel.text =  "\(self.index + 1)"
        }
    }
    
    fileprivate lazy var thumbnailImageView: UIImageView = {
        let thumbnailImageView = UIImageView()
        thumbnailImageView.contentMode = .scaleAspectFill
        thumbnailImageView.clipsToBounds = true
        
        return thumbnailImageView
    }()
    
    fileprivate let checkView = DKImageCheckView()
    
    override var isSelected: Bool {
        didSet {
            checkView.isHidden = !super.isSelected
        }
    }
    
} /* DKAssetGroupDetailCell */
